### PR TITLE
Fixed misuse of delta arg in ChainReader and docs of MDAnalysis.Writer

### PR DESCRIPTION
**Pull request by mnmelo**
If this gets merged do note that there's a mention to the version from which the `delta` keyword was deprecated. It must be updated when the actual version numbering is decided.

### CODE REVIEW

File: package/MDAnalysis/coordinates/base.py [hunk 1]
```diff
@@ -1228,13 +1228,22 @@ def __init__(self, filenames, **kwargs):
...
+        self.__active_reader_index = 0
 
         self.skip = kwargs.get('skip', 1)
-        self._default_delta = kwargs.pop('delta', None)
+        if 'delta' in kwargs:
+            warnings.warn("Keyword 'delta' is now deprecated "
+                          "(from version 0.13); "
+                          "use 'frame_dt' instead", DeprecationWarning)
+        # Give priority to 'frame_delta' over the deprecated 'delta'
+        self._default_delta = kwargs.pop('frame_dt', kwargs.pop('delta', None))
```
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/base.py]: Why can't this be called just `dt`?

**Comment by mnmelo** [comment]
I guess `dt` would work, but might be confusing because it will not necessarily not match what the user gets back when asking for `trajectory.dt` (in which case the current reader's `dt` is returned).

If you think the confusion potential is low, let's go for it.

If we keep `frame_dt`, I'll also add it as an attribute, so that the user can recover it after the `ChainReader` is created.

**Comment by richardjgowers** [comment]
So `frame_dt` sets the `dt` for each Reader in the chainreader, unless the Reader has a better idea?

File: package/MDAnalysis/coordinates/base.py (hunk 1 above)
Comment (orbeckst) [inline package/MDAnalysis/coordinates/base.py]: I'd also prefer not to introduce yet another variation of delta/dt to keep the interface reasonably consistent.

**Comment by mnmelo** [comment]
`frame_dt` is only used when `trajectory.time` is asked for. The Readers keep their own `dt` and will never be otherwise replaced by `frame_dt` (`trajectory.dt` returns the `dt` of the underlying reader, which may differ from `frame_dt`). This is the same behavior we already had. My modifications only change the keyword from `delta` to `frame_dt`.

The right way for the ChainReader to get the current trajectory time would be to know the total time duration and dt of every sub trajectory. The current traj time would then be the sum of traj durations, for trajs before the current, plus the current dt times the number of frames elapsed in the current traj.

This is not yet implemented, and the `delta`/`frame_dt` approach works around it. Even in the case when the right behavior is implemented it might be good to have `frame_dt` as a means to force a common ChainReader dt.

**Comment by dotsdl** [comment]
Related, for reference: #309

**Comment by mnmelo** [comment]
Ok, I'll use `dt`, then. Shall I change the behavior of `trajectory.dt` to point to this `dt` instead? Or do we keep current behavior?

**Comment by richardjgowers** [comment]
So current behaviour, (which I'd rather not change), for normal Readers

 - `dt` should "belong" to the `Timestep` object
 - `Reader.dt` must just point to `Reader.ts.dt`.
 - Any `dt` kwarg to the `Reader` gets passed to the `Timestep` as part of the `_ts_kwargs` dict that all `Readers` have.
 - If a `Timestep` has a kwarg provided `dt` it overrides anything else.
 - `Reader.time` points to `Reader.ts.time` (time also belongs to Timesteps)

If the `chainreader` passes its `dt` kwargs to each child `Reader`, would this solve this issue?

**Comment by mnmelo** [comment]
We could make it as you say, and basically clobber any `dt` information in the trajectories.
As it is right now, the user can still access the `dt` of the underlying trajectories if they wish. With your solution they'd all be replaced by the same value.

Everyone in favor of this?

**Comment by mnmelo** [comment]
Ok, since it wasn't that much work I implemented the proper time counting in the ChainReader, which will take into account each sub-reader's `dt`. As per @richardjgowers suggestion, this can be overridden by passing `dt` to the ChainReader, which will then pass it to each sub-reader instance so that they all have the same `dt`.

**Comment by orbeckst** [comment]
That seems to be a good solution!

**Comment by richardjgowers** [comment]
Yeah I think ChainReader definitely needs some more attention in the future, but this fixes the issue at hand

**Pull request merged**